lib/Partitioner/Partitioner.cpp

```diff
@@ -81,6 +81,95 @@ void Partitioner::initOpMemUsage() {
   }
 }
 
+/// Get the minimal compute time for each op in the function.
+void Partitioner::initOpComputeTime() {
+  computeTime_.clear();
+
+  // This code assumes all ops are BW limited from SRAM; except
+  // if the input does not fit in SRAM -- then it is DRAM BW limited
+  float peakDramBw = deviceInfo_[0].peakDramBw;
+  float peakSramBw = deviceInfo_[0].peakSramBw;
+  size_t sramCapacity = deviceInfo_[0].sramCapacity;
+  float peakCompute = deviceInfo_[0].peakCompute;
+
+  for (auto &node : F_->getNodes()) {
+    /// compute memory side bytes for inputs from DRAM, SRAM.
+    int n = node.getNumInputs();
+    uint64_t size_dram = 0;
+    uint64_t size_sram = 0;
+    if (node.getKind() == Kinded::Kind::SaveNodeKind) {
+      computeTime_[&node] = 0.0f;
+      continue;
+    }
+
+    for (int i = 0; i < n; i++) {
+      auto ty = node.getNthInput(i).getNode()->getType(0);
+      uint64_t size_input = ty->getSizeInBytes();
+      if (size_input > sramCapacity) {
+        size_dram += size_input;
+      } else {
+        size_sram += size_input;
+      }
+    }
+
+    // Repeat for outputs
+    if (node.getNumResults() > 0) {
+      auto myty = node.getType(0);
+      uint64_t size_output = myty->getSizeInBytes();
+      if (size_output > sramCapacity) {
+        size_dram += size_output;
+      } else {
+        size_sram += size_output;
+      }
+    }
+
+    /// Assume actual compute is only for Matmuls.
+    /// TODO: fix this assumption later (FC, Convolutions, BatchMatmul)
+    uint64_t total_ops = 0;
+    if (node.getKind() == Kinded::Kind::MatMulNodeKind) {
+      auto *MMN = llvm::dyn_cast<MatMulNode>(&node);
+      auto lhs_dims = MMN->getLHS().dims();
+      auto rhs_dims = MMN->getRHS().dims();
+      total_ops = lhs_dims[0] * lhs_dims[1] * rhs_dims[1];
+    }
+
+    /// Compute compute roofline as max of flops, DRAM, SRAM BW
+    computeTime_[&node] = std::max(
+        total_ops * 1.0f / peakCompute,
+        std::max(size_dram * 1.0f / peakDramBw, size_sram * 1.0f / peakSramBw));
+  }
+}
+
+/// Get the minimal communication time for each edge in the function.
+void Partitioner::initEdgeCommTime() {
```

Comment:
I am not sure if we do need this initialization. The computation of the communication is pretty easy : output_size / peakPCIeBw and we can get it when we need it. Therefore, I don't think we need a map to store all the values.

Reply:
Ok, I think I agree with that.